core/src/main/scala/akka/persistence/r2dbc/internal/BySliceQuery.scala

```diff
@@ -303,11 +305,30 @@ import org.slf4j.Logger
           throw new IllegalArgumentException(
             s"Unexpected offset [$offset] before latestBacktracking [${state.latestBacktracking}].")
 
-        state.copy(latestBacktracking = offset, rowCount = state.rowCount + 1)
+        // TODO: Could be nicer to track the seen count in TimestampOffset itself, along with the latest seen map.
+        //       Either as a seen count directly, or as a range of seen sequence numbers for each persistence id.
```

Comment:
I'm not convinced that would be better, since TimestampOffset is in Akka core and this is rather deep into implementation details of this query.

Reply:
Yeah, very specifically used here for the backtracking. I was thinking if it was added to TimestampOffset, then it could be more general, in case it's useful elsewhere for similar reasons. There's the `seen` map already, with the latest sequence number. Could be an additional `firstSeen` map, so that it stays compatible, and either the seen counts, or the range of sequence numbers, can be derived from the two maps. Otherwise, if not applicable elsewhere, can just remove the comment too.

Reply:
Let's keep it in the internal state for now (remove the todo). If we need it elsewhere we can revisit.
It would raise questions like, should the offset store also be able to recreate that firstSeen when the offset is loaded at startup (as far as I see we don't need that).

Reply:
Right, should not be needed here — for the offset store to load that information. If a loaded offset was being used to initiate a query that was tracking this, then a loaded firstSeen would be useful, as the loaded seen map would be used for filtering. But here the first backtracking offset is created fresh, starting with an empty seen.